atomisticparsers/gromacs/parser.py

```diff
@@ -83,27 +83,46 @@ def str_to_input_parameters(val_in):
                     parameters[val_array.group(1)].append(value[0] if len(value) == 1 else value)
             return parameters
 
-        def str_to_energies(val_in):
-            energy_keys_re = re.compile(r'(.+?)(?:  |\Z| P)')
-            keys = []
-            values = []
-            energies = dict()
-            for val in val_in.strip().split('\n'):
-                val = val.strip()
-                if val[0].isalpha():
-                    keys = [k.strip() for k in energy_keys_re.findall(val)]
-                    keys = ['P%s' % k if k.startswith('res') else k for k in keys if k]
+        def str_to_energies(energies_block):
+
+            max_char_name = 14
+            rows = energies_block.strip().split("\n")
+            names = [rows[i_row] for i_row in range(len(rows)) if i_row % 2 == 0]
+            values = [rows[i_row] for i_row in range(len(rows)) if i_row % 2 != 0]
+
+            names_re = r"(?<!\S)[A-Za-z0-9.()\/-]{1,}(?: [\S][A-Za-z0-9.()\/-]{1,})*(?!\S)"
+            values_re = r"[-+]?\d+\.\d+e[+-]?\d+"
+            names = [re.findall(names_re, name) for name in names]
+            names = [name for group in names for name in group]
+            names_split = []
+            for name in names:
+                if len(name) <= max_char_name:
+                    names_split.append(name.strip())
                 else:
-                    values = val.split()
-                    for n, key in enumerate(keys):
-
-                        if key == 'Temperature':
-                            energies[key] = float(values[n]) * ureg.kelvin
-                        elif key.startswith('Pres'):
-                            key = key.rstrip(' (bar)')
-                            energies[key] = float(values[n]) * ureg.bar
-                        else:
-                            energies[key] = float(values[n]) / MOL * ureg.kJ
+                    remainder_len = len(name) % max_char_name
```

Comment:
I am not very happy with this. I think we should simply do 
```python
        def str_to_energies(val_in):
            n_chars_val = len(re.search(r'( +Total Energy)', val_in).group(1))
            energies = {}
            rows = [v for v in val_in.splitlines() if v]
            for n in range(0, len(rows), 2):
                pointer = 0
                while pointer < len(rows[n]):
                    key = rows[n][pointer: pointer + n_chars_val].strip()
                    value = rows[n + 1][pointer: pointer + n_chars_val]
                    if key == 'Temperature':
                        energies[key] = float(value) * ureg.kelvin
                    elif key.startswith('Pres'):
                        key = key.rstrip(' (bar)')
                        energies[key] = float(value) * ureg.bar
                    else:
                        energies[key] = float(value) / MOL * ureg.kJ
                    pointer += n_chars_val
        return energies
```
also change the regex for the energies to 
```python
            Quantity(
                'energies',
                r'Energies \(kJ/mol\).*(\s*[\s\S]+?)\n\n',

Reply:
To the later point, I just found there was an error in my regex, I just changed it to 
```
            Quantity(
                'energies',
                r'Energies \(kJ/mol\)\s*([\s\S]+?)\n(?:D.* step.* load imb.*|\n\n)',
                str_operation=str_to_energies, convert=False),
```

But I had to add this other term because of some printing "errors" that I found in some of my test log files, e.g.:
```
           Step           Time         Lambda
           4000        8.00000        0.00000

   Energies (kJ/mol)
        LJ (SR)        LJ (LR)   Coulomb (SR)   Coulomb (LR)      Potential
   -1.00701e+03    0.00000e+00    0.00000e+00    0.00000e+00   -1.00701e+03
    Kinetic En.   Total Energy    Temperature Pressure (bar)
    3.96211e+03    2.95511e+03    3.18005e+02    5.51459e+03
DDD  step 4499 load imb.: force  5.1%
```

This can happen where the load imb. line is printed directly under the energy block. Obv. I am open to suggestions if you have a more general regex to avoid grabbing this line.

Reply:
Ah yes, I have completely forgotten about this, please change it to 
```python
Quantity(
                'energies',
                r'Energies \(kJ/mol\).*\n(\s*[\s\S]+?)(?:\n.*step.* load imb.*|\n\n)',

Reply:
This will not cause any problem since it is not captured. We can simply keep on expanding the regex if we find more cases.

Reply:
Actually, my new regex also doesn't work. I find it extremely difficult to implement regex into the parser because the functionality of the regex expression is somehow different than the online tool that I use for testing regex. I think it's something to do with how line breaks are dealt with but I am not sure. If you have any advice about this, it would be much appreciated.

Reply:
I think we change \n to [\n\r] (see my other comment) because windows print \r.